### PR TITLE
Make clang-format errors more helpful

### PR DESCRIPTION
**Pull request by mlevesquedion**
Specifically:
- Provide a command that can be used to fix formatting problems automatically
- Make it obvious what version of clang-format to use

Also:
- Don't specify --style=google because we have a .clang-format file that already specifies that.
- Only show the version if there are formatting failures

Closes https://github.com/openxla/stablehlo/issues/2223.

### CODE REVIEW

File: build_tools/github_actions/lint_clang_format.sh [hunk 1]
```diff
@@ -51,7 +49,12 @@ fi
 echo "Running clang-format [mode=$FORMAT_MODE]..."
 echo "  Files: " "${CHANGED_FILES[@]}"
 if [[ $FORMAT_MODE == 'fix' ]]; then
-  clang-format --style=google -i "${CHANGED_FILES[@]}"
+  clang-format -i "${CHANGED_FILES[@]}"
```
Comment (GleasonK) [inline build_tools/github_actions/lint_clang_format.sh:52]: why get rid of the `--style`? I'm forgetting why it was added but I think there were issues when removing this.. can you try running this clang format command on the repo and see how much changes?

**Review by GleasonK — APPROVED** [review]

File: build_tools/github_actions/lint_clang_format.sh [hunk 2]
```diff
@@ -51,7 +49,12 @@ fi
...
 else
-  clang-format --style=google --dry-run --Werror "${CHANGED_FILES[@]}"
+  clang-format --dry-run --Werror "${CHANGED_FILES[@]}" || {
+      echo "Please format your code before pushing:"
+      echo "  ./build_tools/github_actions/lint_clang_format.sh -f"
+      echo "Make sure you are using the right major version of clang-format:"
+      echo "  $(clang-format --version)"
+    }
```
Comment (GleasonK) [inline build_tools/github_actions/lint_clang_format.sh:59]: Can you try making this fail? I'm curious if `echo` being the last command will make it seem like the job actually passed in GH Actions.

File: build_tools/github_actions/lint_clang_format.sh (hunk 1 above)
Comment (mlevesquedion) [reply build_tools/github_actions/lint_clang_format.sh:52]: We have a `.clang-format` file which specifies `BasedOnStyle: Google`, so the flag is not needed.

File: build_tools/github_actions/lint_clang_format.sh (hunk 2 above)
Comment (mlevesquedion) [reply build_tools/github_actions/lint_clang_format.sh]: It absolutely will! Thanks, good catch.

**Pull request merged**